plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -619,6 +621,49 @@ public void testExternalTableFilesRetainedOnDrop()
         assertEquals(getTableFiles(tableName).size(), fileCount);
     }
 
+    @Test
+    public void testRenameExternalTable()
+    {
+        String oldTable = "test_external_table_rename_old_" + randomTableSuffix();
+        String tableLocation = getLocationForTable(bucketName, oldTable);
+        assertUpdate(
+                format("CREATE TABLE %s (a bigint, b double) WITH (location = '%s')", oldTable, tableLocation));
+
+        String newTable = "test_rename_new_" + randomTableSuffix();
+        assertUpdate("ALTER TABLE " + oldTable + " RENAME TO " + newTable);
+
+        assertThat(query("SHOW TABLES LIKE '" + oldTable + "'"))
+                .returnsEmptyResult();
+        assertThat(query("SELECT a, b FROM " + newTable))
+                .returnsEmptyResult();
+
+        assertUpdate("INSERT INTO " + newTable + " (a, b) VALUES (42, -38.5)", 1);
+        assertThat(query("SELECT CAST(a AS bigint), b FROM " + newTable))
+                .matches("VALUES (BIGINT '42', -385e-1)");
+
+        assertUpdate("DROP TABLE " + newTable);
+        assertThat(query("SHOW TABLES LIKE '" + newTable + "'"))
+                .returnsEmptyResult();
+    }
+
+    @Test
+    public void testRenameManagedTableIsNotSupportedWhenRenameManagedTablesNotEnabled()
+            throws Exception
+    {
+        QueryRunner queryRunner = createDeltaLakeQueryRunner(
+                ImmutableMap.<String, String>builder()
+                        .put("delta.metadata.cache-ttl", TEST_METADATA_CACHE_TTL_SECONDS + "s")
+                        .put("hive.metastore-cache-ttl", TEST_METADATA_CACHE_TTL_SECONDS + "s")
+                        .buildOrThrow());
+
+        String tableName = "test_rename_" + randomTableSuffix();
+        QueryAssertions.assertUpdate(queryRunner, queryRunner.getDefaultSession(), "CREATE TABLE " + tableName + " AS SELECT 123 x", OptionalLong.of(1), Optional.empty());
+
+        String renamedTable = "test_rename_new_" + randomTableSuffix();
+        QueryAssertions.assertQueryFails(queryRunner, getSession(), "ALTER TABLE " + tableName + " RENAME TO " + renamedTable, "Renaming managed tables will move the data on the underlying storage system. Set 'delta.rename-managed-tables.enabled' to enable this feature.");
```

Comment:
You can use here `assertThatThrownBy` instead of changing the visibility of the method `QueryAssertions.assertQueryFails`

Reply:
Would work, but isn't changing the visibility the better option here? Why is `public static void assertUpdate(QueryRunner queryRunner, Session session, @Language("SQL") String sql, OptionalLong count, Optional<Consumer<Plan>> planAssertion)` public? Changing the visibility would allow tests to be colocated to test features that require multiple queryrunners and have a similar recognizable API. Other alternative would be to move it to a separate test.

Reply:
`assertThatThrownBy(() -> query)` or leverage inherited `AbstractTestQueryFramework#assertQueryFails`.

(the fact that you couldn't make a static import could ring a bell:)

Reply:
The reason why I'm doing this is because I create **another** QueryRunner in this specific test without the setting `delta-rename-managed-tables.enabled=true`. That's why i can't simply call the standard methods. 

I will move this test to a separate class.